src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSupervisor.scala

```diff
@@ -134,20 +149,55 @@ class ClusterMonitorSupervisor(monitorConfig: MonitorConfig, dataprocConfig: Dat
     }
   }
 
-  def autoFreezeClusters: Future[Unit] = {
+  def autoFreezeClusters(): Future[Unit] = {
     val clusterList = dbRef.inTransaction {
       _.clusterQuery.getClustersReadyToAutoFreeze()
     }
     clusterList map { cl =>
       cl.foreach { c =>
         logger.info(s"Auto freezing cluster ${c.clusterName} in project ${c.googleProject}")
-        leoService.internalStopCluster(c).failed.foreach { e =>
+        leonardoService.internalStopCluster(c).failed.foreach { e =>
           logger.warn(s"Error occurred auto freezing cluster ${c.projectNameString}", e)
         }
       }
     }
   }
 
+  private def createClusterMonitors(implicit executionContext: ExecutionContext): Unit = {
+    dbRef
+      .inTransaction { _.clusterQuery.listMonitoredFullCluster() }
+      .onComplete {
+        case Success(clusters) =>
+          val clustersNotAlreadyBeingMonitored = clusters.toSet -- monitoredClusters.value
+
+          clustersNotAlreadyBeingMonitored foreach {
+            case c if c.status == ClusterStatus.Deleting => {
+              logger.info("deleting cluster")
+              self ! ClusterDeleted(c)
+            }
+            case c if c.status == ClusterStatus.Stopping => {
+              logger.info("stopping cluster")
+              self ! ClusterStopped(c)
+            }
+            case c if c.status == ClusterStatus.Starting => self ! ClusterStarted(c)
+
+            case c if c.status == ClusterStatus.Updating => self ! ClusterUpdated(c)
+
+            case c => {
```

Comment:
Minor, maybe have an explicit case for `Creating`:
```
case c if c.status == ClusterStatus.Creating =>
```
And the default case could log a warning "unhandled status" or something like that.